### PR TITLE
Document --single-sensor-mode st2sensorcontainer service CLI option

### PR DESCRIPTION
**Pull request by Kami**
This pull request documents new ``--single-sensor-mode`` sensor container CLI option.

(Will cherry pick it in v2.9 branch once it's merged)

### CODE REVIEW

**Comment by armab** [comment]
I think this documentation might be a better fit for "Sensors" or "Sensors Partitioning" document and just provide a reference to that doc from HA.

**Comment by Kami** [comment]
@armab Yeah, I was wondering where should I put it - either HA or sensor partitioning section.

@LindsayHill what do you think?

**Comment by LindsayHill** [comment]
Good point @armab - at a minimum it needs cross-linking.

I think it makes sense to have a mention of it in HA docs, and then config details in the sensor partitioning doc.

**Comment by Kami** [comment]
I had a look at sensor partitioning docs again and I don't think it belongs there.

Sensor partitioning and this flag are two unrelated functionalities. Operators who will use ``--single-sensor-mode`` flag will likely (need to) handle sensor partitioning themselves (they will need to manually assign which sensor runs on which node as per discussion with @armab in the past) and not utilize the build in partitioning functionality (in fact, those two things are mutually exclusive).

Similar story for general "Sensors" section - this one is geared towards end users and not operators.

For now, I will keep it in HA document, feel free to update it if you think it should go somewhere else.

**Comment by Kami** [comment]
Added a note / cross-reference in 3e6614b.

File: docs/source/reference/ha.rst [hunk 1]
```diff
@@ -77,6 +77,26 @@ implemented with HA in mind so that the same sensor can be deployed on multiple
...
+To account for such deployments, sensor container can be started in single sensor mode using
+``--single-sensor-mode`` and ``--sensor-ref`` command line options. When those options are
+provided, sensor container service will run a single sensor and exit immediately if a sensor
+crashes or similar.
+
+For example:
+
+.. code-block:: bash
+
+  st2sensorcontainer --single-sensor-mode --sensor-ref linux.FileWatchSensor
```
Comment (armab) [inline docs/source/reference/ha.rst]: I think this single-sensor-per-process mode represents manual way of partitioning sensors.

Pretty similar to File-based partitioning https://docs.stackstorm.com/reference/sensor_partitioning.html#file where user defines associations on which node to start this or that sensor.

From this point of view I think it makes sense to move this configuration example from HA to Sensor Partitioning doc.

**Pull request merged**